src/parser/view_transform.c

```diff
@@ -4849,6 +4959,44 @@ mq_push_paths (PARSER_CONTEXT * parser, PT_NODE * statement, void *void_arg, int
   return statement;
 }
 
+/*
+ * mq_rewrite_dblink_as_subquery () - rewrite dblink as a subquery
+ *   return: PT_NODE *
+ *   parser(in): parser environment
+ *   node(in): possible dblink query
+ */
+static PT_NODE *
+mq_rewrite_dblink_as_subquery (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *walk)
+{
+  PT_NODE *spec, *derived = NULL;
+  PT_NODE *derived_table;
+
+  if (node->node_type != PT_SELECT)
+    {
+      return node;
+    }
+
+  for (spec = node->info.query.q.select.from; spec; spec = spec->next)
+    {
+      if ((derived_table = spec->info.spec.derived_table)
+	  && spec->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE)
+	{
+	  derived = mq_rewrite_dblink_as_derived (parser, derived_table);
```

Comment:
How about removing the derived_table variable and doing something like below?
Evaluation through substitution statements like the one below can be misleading.
(derived_table = spec->info.spec.derived_table)

```
if (spec->info.spec.derived_table && 
     spec->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE)
{ 
   derived = mq_rewrite_dblink_as_derived (parser, spec->info.spec.derived_table);

```

Reply:
I agree with you.